src/uu/dd/src/dd.rs

```diff
@@ -0,0 +1,1296 @@
+        i.cflags.unblock.is_some()
+    }
+    // Local Helper Fns -------------------------------------------------
+    #[inline]
+    fn perform_swab(buf: &mut [u8]) {
+        let mut tmp;
+
+        for base in (1..buf.len()).step_by(2) {
+            tmp = buf[base];
+            buf[base] = buf[base - 1];
+            buf[base - 1] = tmp;
+        }
+    }
+    // ------------------------------------------------------------------
+    // Read
+    let mut buf = vec![BUF_INIT_BYTE; bsize];
+    let mut rstat = match i.cflags.sync {
+        Some(ch) => i.fill_blocks(&mut buf, ch)?,
+        _ => i.fill_consecutive(&mut buf)?,
+    };
+    // Return early if no data
+    if rstat.reads_complete == 0 && rstat.reads_partial == 0 {
+        return Ok((rstat, buf));
+    }
+
+    // Perform any conv=x[,x...] options
+    if i.cflags.swab {
+        perform_swab(&mut buf);
+    }
+    if is_conv(i) || is_block(i) || is_unblock(i) {
+        let buf = conv_block_unblock_helper(buf, i, &mut rstat)?;
+        Ok((rstat, buf))
+    } else {
+        Ok((rstat, buf))
+    }
+}
+
+fn print_io_lines(update: &ProgUpdate) {
+    eprintln!(
+        "{}+{} records in",
+        update.reads_complete, update.reads_partial
+    );
+    if update.records_truncated > 0 {
+        eprintln!("{} truncated records", update.records_truncated);
+    }
+    eprintln!(
+        "{}+{} records out",
+        update.writes_complete, update.writes_partial
+    );
+}
+fn make_prog_line(update: &ProgUpdate) -> String {
+    let btotal_metric = Byte::from_bytes(update.bytes_total)
+        .get_appropriate_unit(false)
+        .format(0);
+    let btotal_bin = Byte::from_bytes(update.bytes_total)
+        .get_appropriate_unit(true)
+        .format(0);
+    let safe_millis = cmp::max(1, update.duration.as_millis());
+    let xfer_rate = Byte::from_bytes(1000 * (update.bytes_total / safe_millis))
+        .get_appropriate_unit(false)
+        .format(1);
+
+    format!(
+        "{} bytes ({}, {}) copied, {:.1} s, {}/s",
+        update.bytes_total,
+        btotal_metric,
+        btotal_bin,
+        update.duration.as_secs_f64(),
+        xfer_rate
+    )
+}
+fn reprint_prog_line(update: &ProgUpdate) {
+    eprint!("\r{}", make_prog_line(update));
+}
+fn print_prog_line(update: &ProgUpdate) {
+    eprintln!("{}", make_prog_line(update));
+}
+fn print_xfer_stats(update: &ProgUpdate) {
+    print_io_lines(update);
+    print_prog_line(update);
+}
+
+/// Generate a progress updater that tracks progress, receives updates, and responds to signals.
+fn gen_prog_updater(rx: mpsc::Receiver<ProgUpdate>, xfer_stats: Option<StatusLevel>) -> impl Fn() {
+    // --------------------------------------------------------------
+    fn posixly_correct() -> bool {
+        env::var("POSIXLY_CORRECT").is_ok()
+    }
+    // --------------------------------------------------------------
+    move || {
+        const SIGUSR1_USIZE: usize = signal::SIGUSR1 as usize;
```

Comment:
It looks like `SIGUSR1` is not available on windows.
Where are we receiving `SIGUSR1` signals from?

Reply:
Good call. I've added a guard to the relevant code so it's only present on Linux.